src/components/EditProfileForm/EditProfileForm.tsx

```diff
@@ -1,10 +1,122 @@
+import Icon from '../Icon';
+import Input from '../Input';
+import Spinner from '../Spinner';
 import styles from './EditProfileForm.module.css';
 
 const EditProfileForm = () => {
+  const isLoading = useAppSelector(selectIsLoading);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const { username, email, avatar } = useAppSelector(selectUser);
+  const { id } = useAppSelector(selectToken);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<EditProfilePropsType>({
+    resolver: yupResolver(editProfileSchema),
+    defaultValues: {
+      username,
+      email,
+    },
+  });
+
+  const onSubmit: SubmitHandler<EditProfilePropsType> = async (data) => {
+    const formData = createFormData({
+      data,
+      omitFields: ['username', 'avatar'],
+    });
+    formData.append('name', data.username);
+    data.avatar.length && formData.append('avatar', data.avatar[0]);
+
+    /// @ts-expect-error In fact formData is EditProfilePropsType object
+    dispatch(updateUser({ id, credentials: formData }));
+  };
+
   return (
-    <section className={styles.form}>
-      <form></form>
-    </section>
+    <form onSubmit={handleSubmit(onSubmit)} encType="multipart/form-data">
+      <div className={styles.inputsContainer}>
+        <AvatarInput
+          register={register('avatar')}
+          imagePath={avatar}
+          imageContainerStyles={styles.avatarContainer}
+          buttonTitle="Edit photo"
+          buttonStyles={styles.avatarButton}
+        />
+        <div className={styles.userDecrContainer}>
+          <Input
+            register={register('username')}
+            title="Full Name"
+            placeholder="Full Name"
+            error={errors.username?.message}
+          />
+
+          <Input
+            register={register('email')}
+            title="E-mail"
+            placeholder="example@gmail.com"
+            error={errors.email?.message}
+            disabled={true}
+          />
+        </div>
+      </div>
+
+      <div className={styles.buttonsGroup}>
+        <Button
+          type="submit"
+          size="lg"
+          variant="primary"
+          disabled={isLoading}
+          className={styles.loginButton}
+          icon={
+            isLoading ? <Spinner containerClassName={styles.spinner} /> : null
+          }
+        >
+          Submit
+        </Button>
+        <Button
+          type="button"
+          variant="link"
+          icon={<Icon id="lock" />}
+          iconPosition="before"
+          onClick={() => {}}
+        >
+          Change password
+        </Button>
+        <Button
+          type="button"
+          variant="link"
+          icon={<Icon id="trash" />}
+          iconPosition="before"
+          onClick={async () => {
+            const { payload } = await dispatch(deleteUser(id));
+            payload === true && navigate(ROUTES.REGISTER);
+          }}
+          className={styles.trashButton}
```

Comment:
Better write function for onClick cb if they more than one line long. I makes our jsx layout more clear